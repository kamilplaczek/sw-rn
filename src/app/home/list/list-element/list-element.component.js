import React, { Component } from 'react';
import { styles } from './list-element.styles';
import { Text, TouchableHighlight, View } from 'react-native';

export class ListElement extends Component {
  render() {
    const { film, onPress } = this.props;

    return (
      <TouchableHighlight testID="listElement" underlayColor="transparent" onPress={onPress}>
        <View style={styles.container}>
          <View style={styles.iconContainer}>
            <Text style={styles.icon}>ðŸŽž</Text>
          </View>
          <View style={styles.content}>
            <Text style={styles.title}>{film.title}</Text>
            <Text>{film.director}</Text>
            <Text>{film.releaseDate}</Text>
          </View>
        </View>
      </TouchableHighlight>
    );
  }
}
