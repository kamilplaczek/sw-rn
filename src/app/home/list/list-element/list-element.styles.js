import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    borderBottomColor: '#ccc',
    borderBottomWidth: 1,
    paddingBottom: 10,
    paddingTop: 10,
  },
  iconContainer: {
    width: 70,
    alignItems: 'center',
    justifyContent: 'center',
  },
  icon: {
    fontSize: 30,
  },
  content: {
    flex: 1,
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
  },
});
