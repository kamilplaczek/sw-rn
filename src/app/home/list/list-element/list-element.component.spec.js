import * as React from 'react';
import renderer from 'react-test-renderer';
import { ListElement } from './list-element.component';
import { shallow } from 'enzyme';

describe('list element component', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <ListElement
          film={{
            name: 'foo',
            director: 'bar',
            releaseDate: '2018-09-20',
          }}
          onPress={() => {}}
        />
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('calls onPress handler', () => {
    const onPress = jest.fn();
    const wrapper = shallow(
      <ListElement
        film={{
          name: 'foo',
          director: 'bar',
          releaseDate: '2018-09-20',
        }}
        onPress={onPress}
      />
    );
    wrapper.find({ testID: 'listElement' }).simulate('press');
    expect(onPress).toBeCalled();
  });
});
