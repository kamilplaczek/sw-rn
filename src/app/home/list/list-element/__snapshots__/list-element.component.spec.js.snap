// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list element component renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="listElement"
>
  <View
    style={
      Object {
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 70,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
          }
        }
      >
        ðŸŽž
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      />
      <Text>
        bar
      </Text>
      <Text>
        2018-09-20
      </Text>
    </View>
  </View>
</View>
`;
